### PR TITLE
feat: return asset ids in new resolver appendManyAssets

### PR DESCRIPTION
**Pull request by theodu**

### CODE REVIEW

File: src/kili/services/asset_import/base.py [hunk 1]
```diff
@@ -459,9 +468,10 @@ def import_assets_by_batch(
...
 
-        responses = []
+        created_asset_ids = []
         for i, batch_assets in enumerate(batch_generator):
             # check last batch only
             verify = i == (len(batch_generator) - 1) and self.verify
-            responses.append(batch_importer.import_batch(batch_assets, verify))
-        return responses[-1]
+            batch_created_asset_ids = batch_importer.import_batch(batch_assets, verify)
+            created_asset_ids.append(batch_created_asset_ids)
```
Comment (PierreLeveau) [inline src/kili/services/asset_import/base.py:476]: extend instead  of append? We want a flat list imho

Comment (PierreLeveau) [reply src/kili/services/asset_import/base.py:476]: maybe worth doing a test by mocking the resolver answer

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: src/kili/services/asset_import/base.py (hunk 1 above)
Comment (theodu) [reply src/kili/services/asset_import/base.py:476]: yes you are right!

File: src/kili/entrypoints/mutations/asset/__init__.py [hunk 2]
```diff
@@ -140,14 +140,14 @@ def append_many_to_dataset(
...
-        result = import_assets(
+        created_assets_ids = import_assets(
             self,
             project_id=project_id,
             assets=assets,
             disable_tqdm=disable_tqdm,
             verify=wait_until_availability,
         )
-        return result
+        return {"id": project_id, "created_assets_ids": created_assets_ids}
```
Comment (Jonas1312) [inline src/kili/entrypoints/mutations/asset/__init__.py:150]: I would just name the key `asset_ids`

File: src/kili/entrypoints/mutations/asset/__init__.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def append_many_to_dataset(
         json_metadata_array: Optional[List[dict]] = None,
         disable_tqdm: bool = False,
         wait_until_availability: bool = True,
-    ) -> Optional[Dict[Literal["id"], str]]:
+    ) -> Optional[Dict[Literal["id", "created_assets_ids"], Union[str, List[str]]]]:
         # pylint: disable=line-too-long
         """Append assets to a project.
```
Comment (Jonas1312) [inline src/kili/entrypoints/mutations/asset/__init__.py:55]: need to change the docstring `Returns: ...`

File: src/kili/services/asset_import/base.py [hunk 4]
```diff
@@ -86,16 +87,20 @@ def __init__(self, kili, project_params: ProjectParams, batch_params: BatchParam
         self.logger.setLevel(logging.INFO)
 
     @pagination.api_throttle
```
Comment (Jonas1312) [inline src/kili/services/asset_import/base.py:89]: I thought I had removed all api_throttle, maybe you need to rebase on main

File: src/kili/entrypoints/mutations/asset/__init__.py [hunk 5]
```diff
@@ -140,14 +140,14 @@ def append_many_to_dataset(
         for key, value in field_mapping.items():
             if value is not None:
                 assets = [{**assets[i], key: value[i]} for i in range(nb_data)]
-        result = import_assets(
+        created_assets_ids = import_assets(
```
Comment (Jonas1312) [inline src/kili/entrypoints/mutations/asset/__init__.py:143]: so for async imports the asset_ids would be empty?  if yes, we should put it in the docstring

**Review by Jonas1312 — CHANGES_REQUESTED** [review]

File: src/kili/entrypoints/mutations/asset/__init__.py (hunk 2 above)
Comment (theodu) [reply src/kili/entrypoints/mutations/asset/__init__.py:150]: it was to make sure that the client understands that we don't return all the project asset ids

Comment (theodu) [reply src/kili/entrypoints/mutations/asset/__init__.py:150]: But I am also good with just asset_ids

Comment (Jonas1312) [reply src/kili/entrypoints/mutations/asset/__init__.py:150]: ah ok makes sense, you can keep it as is then

Comment (theodu) [reply src/kili/entrypoints/mutations/asset/__init__.py]: I used asset_ids, I also hesitated to put it as so

**Review by Jonas1312 — APPROVED** [review]

File: tests/services/asset_import/test_import_common.py [hunk 6]
```diff
@@ -55,6 +57,21 @@ def test_cannot_upload_raw_text_to_pdf_project(self, *_):
         with self.assertRaises(FileNotFoundError):
             import_assets(self.kili, self.project_id, assets, disable_tqdm=True)
 
+    @patch.object(ProjectQuery, "__call__", side_effect=mocked_project_input_type("TEXT"))
+    @patch.object(AssetQuery, "count", return_value=1)
+    def test_return_the_ids_of_created_assets(self, *_):
+        assets = [{"content": "One"}, {"content": "Two"}, {"content": "Three"}]
+
+        def graphql_execute_side_effect(*args, **kwargs):
```
Comment (PierreLeveau) [inline tests/services/asset_import/test_import_common.py:65]: this is a hack until we get the new architecture in place correct?

Comment (theodu) [reply tests/services/asset_import/test_import_common.py:65]: yes, it will be much easier after!

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**